src/spikeinterface/generation/hybrid_tools.py

```diff
@@ -0,0 +1,518 @@
+        possoble based on the min_displacement and max_displacement constraints.
+        This avoids a bias in moving templates towards the center of the probe.
+    depth_direction : "x" | "y", default: "y"
+        The direction in which to move the templates. Can be "x" or "y"
+    seed : int or None, default: None
+        Seed for random initialization.
+
+
+    Returns
+    -------
+    Templates
+        The moved templates
+    """
+    seed = _ensure_seed(seed)
+
+    extremum_channel_indices = list(get_template_extremum_channel(templates, outputs="index").values())
+    extremum_channel_indices = np.array(extremum_channel_indices, dtype=int)
+    depth_dimension = ["x", "y"].index(depth_direction)
+    channel_depths = templates.get_channel_locations()[:, depth_dimension]
+    unit_depths = channel_depths[extremum_channel_indices]
+
+    assert margin >= 0, "margin should be positive"
+    top_margin = np.max(channel_depths) + margin
+    bottom_margin = np.min(channel_depths) - margin
+
+    templates_array_moved = np.zeros_like(templates.templates_array, dtype=templates.templates_array.dtype)
+
+    rng = np.random.default_rng(seed)
+    displacements = rng.uniform(low=min_displacement, high=max_displacement, size=templates.num_units)
+    for i in range(templates.num_units):
+        # by default, displacement is positive
+        displacement = displacements[i]
+        unit_depth = unit_depths[i]
+        if not favor_borders:
+            displacement *= rng.choice([-1.0, 1.0])
+            if unit_depth + displacement > top_margin:
+                displacement = -displacement
+            elif unit_depth - displacement < bottom_margin:
+                displacement = -displacement
+        else:
+            # check if depth is closer to top or bottom
+            if unit_depth > (top_margin - bottom_margin) / 2:
+                # if over top margin, move down
+                if unit_depth + displacement > top_margin:
+                    displacement = -displacement
+            else:
+                # if within bottom margin, move down
+                if unit_depth - displacement >= bottom_margin:
+                    displacement = -displacement
+        displacement_vector = np.zeros(2)
+        displacement_vector[depth_dimension] = displacement
+        templates_array_moved[i] = move_dense_templates(
+            templates.templates_array[i][None],
+            displacements=displacement_vector[None],
+            source_probe=templates.probe,
+        )[0]
+
+    return Templates(
+        templates_array=templates_array_moved,
+        sampling_frequency=templates.sampling_frequency,
+        nbefore=templates.nbefore,
+        sparsity_mask=templates.sparsity_mask,
+        channel_ids=templates.channel_ids,
+        unit_ids=templates.unit_ids,
+        probe=templates.probe,
+    )
+
+
+def generate_hybrid_recording(
+    recording,
+    motion_info=None,
+    num_units=10,
```

Comment:
Do we need this when templates is given ?

Reply:
move to generate_sorting_kwargs